provider/scaleway/scaleway_discover.go

```diff
@@ -28,54 +39,151 @@ func (p *Provider) Addrs(args map[string]string, l *log.Logger) ([]string, error
 	if err != nil {
 		return nil, fmt.Errorf("discover-scaleway: %s", err)
 	}
 
-	// Filter servers by tag
 	var addrs []string
-	if servers != nil {
-		for _, server := range servers {
-			if stringInSlice(tagName, server.Tags) {
-				l.Printf("[DEBUG] discover-scaleway: Found server (%s) - %s with private IP: %s",
-					server.Name, server.Hostname, server.PrivateIP)
-				addrs = append(addrs, server.PrivateIP)
+	for _, server := range servers.Servers {
+		a := ""
+		switch addrType {
+
+		case "public_v4":
+			if server.PublicIP != nil {
+				a = server.PublicIP.Address.String()
+			}
+		case "public_v6":
+			if server.IPv6 != nil {
+				a = server.IPv6.Address.String()
 			}
+
+		case "private_v4":
+			a = *server.PrivateIP
+		}
+
+		if a != "" {
+			addrs = append(addrs, a)
 		}
 	}
 
-	l.Printf("[DEBUG] discover-scaleway: Found ip addresses: %v", addrs)
+	l.Printf("discover-scaleway: Found ip addresses: %v", addrs)
 	return addrs, nil
 }
 
-func stringInSlice(a string, list []string) bool {
-	for _, b := range list {
-		if b == a {
-			return true
+func newClient(args map[string]string, l *log.Logger) (*scw.Client, error) {
+	// By default we set default zone and region to fr-par
+	defaultZoneProfile := &scw.Profile{
+		DefaultRegion: scw.StringPtr(scw.RegionFrPar.String()),
+		DefaultZone:   scw.StringPtr(scw.ZoneFrPar1.String()),
+	}
+
+	// config file loading
+	config, err := scw.LoadConfig()
+	// If the config file do not exist, don't return an error as we may find config in ENV or flags.
+	if _, isNotFoundError := err.(*scw.ConfigFileNotFoundError); isNotFoundError {
+		config = &scw.Config{}
+	} else if err != nil {
+		return nil, err
+	}
+	configProfile := &config.Profile
+
+	// Environment config loading
+	envProfile := scw.LoadEnvProfile()
+
+	// Command line argument loading
+	argsProfile := &scw.Profile{}
+
+	projectID := args["project-id"]
+	if projectID != "" {
+		argsProfile.DefaultProjectID = scw.StringPtr(projectID)
+	}
+
+	secretKey := args["secret-key"]
+	if secretKey != "" {
+		argsProfile.SecretKey = scw.StringPtr(secretKey)
+	}
+
+	zone := args["zone"]
+	if zone != "" {
+		z, err := scw.ParseZone(zone)
+		if err != nil {
+			return nil, fmt.Errorf("error during zone parsing: %s", err)
 		}
+		argsProfile.DefaultZone = scw.StringPtr(z.String())
+	}
+
+	organization := args["organization"]
+	if organization != "" {
+		argsProfile.DefaultOrganizationID = scw.StringPtr(organization)
+	}
+
+	profile := scw.MergeProfiles(defaultZoneProfile, configProfile, envProfile, argsProfile)
+
+	l.Printf("[INFO] discover-scaleway: Organization is %q", *profile.DefaultOrganizationID)
+	l.Printf("[INFO] discover-scaleway: Project is %q", *profile.DefaultProjectID)
+	l.Printf("[INFO] discover-scaleway: Secret is %q", hideSecretKey(*profile.SecretKey))
+	l.Printf("[INFO] discover-scaleway: Zone is %q", *profile.DefaultZone)
+
+	opts := []scw.ClientOption{
+		scw.WithUserAgent(fmt.Sprintf("go-discover")),
```

Comment:
is there a version of go-discover we can add in user agent?

Reply:
Unfortunately not at the moment.

Reply:
@dnephin do you think it could be interesting to add support for a go-discover version?.